traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go

```diff
@@ -138,7 +162,7 @@ func CreateV30(w http.ResponseWriter, r *http.Request) {
 
 	ds := tc.DeliveryServiceV30{}
 	if err := json.NewDecoder(r.Body).Decode(&ds); err != nil {
-		api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("decoding: "+err.Error()), nil)
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, fmt.Errorf("decoding: %w", err), nil)
```

Comment:
> Why the change?

Because it preserves the identity of the underlying error that way. Here's [a playground link to an example program that shows what that means](https://go.dev/play/p/GOkCOgci1lh). It accomplishes the same thing as `errors.New(someString+err.Error())` but without destroying any information.

> ...shouldn't we refactor others in the code base too?

In my opinion, yes, and I always request people do it this way in my reviews. I only changed it in the files I was already editing, though, because this is done untold thousands of times throughout ATC. Nobody would ever review that PR. So I just make smaller changes to files as I work on them.

Reply:
Ok, I like that you are taking care of it in each of your PR but I feel that is a bit of scope creep. Because this is refactor(necessary) but I am not a 100% sure in a PR this big it should have been included. I
f you want, create an issue to tackle this as part of tech-debt and we can take care of it at one go.

Reply:
took it out